### DISCUSSION TITLE
Lesson 12:  Connecting WSL2 to Ganache

### DISCUSSION THREAD

**Discussion by ChristianG77** [smartcontractkit/full-blockchain-solidity-course-py#289] [Q&A] [answered]
Hi,

I'm trying to deploy the box contract at the start of lesson 12 and am getting the error below.

The contract deploys on rinkeby so I think it might be an issue connecting WSL2 to Ganache.

Has anyone come across this problem?

christian@DESKTOP-PN8H8CK:/mnt/c/Users/chris/Dropbox/00_CODE/upgrades$ brownie run scripts/deploy_box.py
Brownie v1.17.0 - Python development framework for Ethereum

UpgradesProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 44, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 93, in launch
    raise RPCProcessError(cmd, uri)
RPCProcessError: Unable to launch local RPC client.
Command: ganache-cli
URI: http://127.0.0.1:8545

**Comment by cromewar** [comment, marked as answer]
Hi @ChristianG77 it seems it's trying to access to windows brownie config, I have some recommendations for you.

1. Make sure you have installed ganache-cli on your WSL: `npm install -g ganache-cli`
2. Move your projects to the Home folder of WSL, this will take the brownie configuration of linux instead of the windows one as it's on the home directory, remember [this](https://dev.to/miftahafina/accessing-wsl2-files-from-windows-file-explorer-308o) guide to access to WSL through Windows File Manager, you may see a `.brownie` folder on the home path: `/home/yourUser/.brownie/`

Tip: you can access to the home directory via console typing `cd`

**Comment by ChristianG77** [comment]
Hi @cromewar thanks for your response.

So I installed ganache-cli on WSL and received the below.  Are any of these errors a significant issue?

![image](https://user-images.githubusercontent.com/91568178/139598912-895e77f1-8a33-4d5d-93da-a6aeebc4a43f.png)

I have my ganache server settings as per the below.

![image](https://user-images.githubusercontent.com/91568178/139599111-38437eb6-5952-40c0-b89a-d95f9471806f.png)

Then I took a copy of the code folder and placed it to \\wsl$\Ubuntu-20.04\home\christian\.brownie\00_CODE\upgrades.  I don't need to delete the folder on Windows, correct?  

I deleted the build, compiled and ran run scripts/deploy_box.py and run scripts/deploy_boxy.py --network rinkeby and received the following.

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie compile
Brownie v1.17.0 - Python development framework for Ethereum

Compiling contracts...
  Solc version: 0.8.9
  Optimizer: Enabled  Runs: 200
  EVM Version: Istanbul
Generating build data...
 - OpenZeppelin/openzeppelin-contracts@4.3.2/Ownable
 - OpenZeppelin/openzeppelin-contracts@4.3.2/ERC1967Proxy
 - OpenZeppelin/openzeppelin-contracts@4.3.2/ERC1967Upgrade
 - OpenZeppelin/openzeppelin-contracts@4.3.2/Proxy
 - OpenZeppelin/openzeppelin-contracts@4.3.2/IBeacon
 - OpenZeppelin/openzeppelin-contracts@4.3.2/Address
 - OpenZeppelin/openzeppelin-contracts@4.3.2/Context
 - OpenZeppelin/openzeppelin-contracts@4.3.2/StorageSlot
 - Box
 - BoxV2
 - ProxyAdmin
 - TransparentUpgradeableProxy

Project has been compiled. Build artifacts saved at /home/christian/.brownie/00_CODE/upgrades/build/contracts

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie run scripts/deploy_box.py
Brownie v1.17.0 - Python development framework for Ethereum

UpgradesProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 44, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 93, in launch
    raise RPCProcessError(cmd, uri)
RPCProcessError: Unable to launch local RPC client.
Command: ganache-cli
URI: http://127.0.0.1:8545
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie run scripts/deploy_box.py --network rinkeby
Brownie v1.17.0 - Python development framework for Ethereum

UpgradesProject is the active project.
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 53, in run
    module = _import_from_path(script)
  File "brownie/project/scripts.py", line 149, in _import_from_path
    _import_cache[import_str] = importlib.import_module(import_str)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
ModuleNotFoundError: No module named 'home.christian.'

**Reply by cromewar** [reply]
Try to select the host name and you will find one with (wsl) on its name, give it a try, anyways I'll do further research on this.

**Reply by cromewar** [reply]
I was searching more info over this and yes, the most likely solution is to change the HOSTNAME and select the one with WSL.

**Reply by ChristianG77** [reply]
Thanks for looking into this!

So would the idea be to do such as this lad in the vid and instead of putting SrcCodes as hostname, put wsl$ as hostname?

Seems to make sense as the folder structure starts with \\wsl$

https://www.youtube.com/watch?v=YJXtJWYLDPo

![image](https://user-images.githubusercontent.com/91568178/140029200-c32c7652-b8a4-4c28-8074-f01d0998ad90.png)

**Reply by cromewar** [reply]
Hello @ChristianG77  of course you can do as the video says, however take in consideration that for now on you would have to do all your development on WSL (if you are making a webpage for example) as the hostname always will point to WSL. I still think that is more simple to just change it on ganache, it shoul look like this:

![image](https://user-images.githubusercontent.com/20199180/140059315-c5778532-e4be-4c31-803d-b5d290c10f95.png)

**Reply by ChristianG77** [reply]
Hi @cromewar.

So I've set my ganache as you have recommended via screen shot below, but I am still getting the same problem as per error futher below.

I access my ganache through what I'm guessing is Windows and this is probably where the issue is as the error says, "URI:  http://127.0.0.1:8545" instead of "URI: http://172.29.64.1" as per your error message.

If that is correct, how do I change the HOSTNAME through WSL?  I think I've followed all the correct steps installing ganache on WSL in the initial setup, but for this part I don't know how to access ganache UI via WSL.

Thanks as always for your time and patience with all these questions.

![image](https://user-images.githubusercontent.com/91568178/140180632-789436fd-cf1d-4ad3-b8cd-28feacc4a930.png)

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie compile
Brownie v1.17.0 - Python development framework for Ethereum

Project has been compiled. Build artifacts saved at /home/christian/.brownie/00_CODE/upgrades/build/contracts
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie run scripts/deploy_box.py
Brownie v1.17.0 - Python development framework for Ethereum

UpgradesProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 44, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 93, in launch
    raise RPCProcessError(cmd, uri)
RPCProcessError: Unable to launch local RPC client.
Command: ganache-cli
URI: http://127.0.0.1:8545

**Reply by cromewar** [reply]
Hello @ChristianG77 , yes you should change also the hostname on your brownie is it's no more 127.0.0.1 use this command:

`brownie networks modify nameOfTheNetwork host=newIP`

Example 

`brownie networks modify ganache-ui host=http://172.29.64.1:7545`

**Reply by ChristianG77** [reply]
Ok so in VS Code using the WSL network I've tired 'brownie networks modify ganache-ui host=http://172.29.64.1:7545' and received 'ValueError:  Network 'ganache-ui' does not exit.

I then did 'npm install -g ganache-cli' which then recommended 'npm audit fix --force' which then recommended 'npm audit fix', but unfortunately after trying again 'brownie networks modify ganache-ui host=http://172.29.64.1:7545' still received 'ValueError:  Network 'ganache-ui'.  Not having much luck on this haha!

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie networks modify ganache-ui host=http://172.29.64.1:7545
Brownie v1.17.0 - Python development framework for Ethereum

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/networks.py", line 68, in main
    fn(*args["<arguments>"])
  File "brownie/_cli/networks.py", line 151, in _modify
    raise ValueError(f"Network '{color('bright magenta')}{id_}{color}' does not exist")
ValueError: Network 'ganache-ui' does not exist
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ npm install -g ganache-cli

changed 6 packages, and audited 102 packages in 2s

2 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (7 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ npm audit fix --force
npm WARN using --force Recommended protections disabled.
npm WARN audit fix ansi-regex@4.1.0 node_modules/ganache-cli/node_modules/ansi-regex
npm WARN audit fix ansi-regex@4.1.0 is a bundled dependency of
npm WARN audit fix ansi-regex@4.1.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix ansi-regex@4.1.0 It cannot be fixed automatically.
npm WARN audit fix ansi-regex@4.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix y18n@4.0.0 node_modules/ganache-cli/node_modules/y18n
npm WARN audit fix y18n@4.0.0 is a bundled dependency of
npm WARN audit fix y18n@4.0.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix y18n@4.0.0 It cannot be fixed automatically.
npm WARN audit fix y18n@4.0.0 Check for updates to the ganache-cli package.
npm WARN audit fix elliptic@6.5.3 node_modules/ganache-cli/node_modules/elliptic
npm WARN audit fix elliptic@6.5.3 is a bundled dependency of
npm WARN audit fix elliptic@6.5.3 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix elliptic@6.5.3 It cannot be fixed automatically.
npm WARN audit fix elliptic@6.5.3 Check for updates to the ganache-cli package.
npm WARN audit fix strip-ansi@5.2.0 node_modules/ganache-cli/node_modules/strip-ansi
npm WARN audit fix strip-ansi@5.2.0 is a bundled dependency of
npm WARN audit fix strip-ansi@5.2.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix strip-ansi@5.2.0 It cannot be fixed automatically.
npm WARN audit fix strip-ansi@5.2.0 Check for updates to the ganache-cli package.
npm WARN audit fix wrap-ansi@5.1.0 node_modules/ganache-cli/node_modules/wrap-ansi
npm WARN audit fix wrap-ansi@5.1.0 is a bundled dependency of
npm WARN audit fix wrap-ansi@5.1.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix wrap-ansi@5.1.0 It cannot be fixed automatically.
npm WARN audit fix wrap-ansi@5.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix string-width@3.1.0 node_modules/ganache-cli/node_modules/string-width
npm WARN audit fix string-width@3.1.0 is a bundled dependency of
npm WARN audit fix string-width@3.1.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix string-width@3.1.0 It cannot be fixed automatically.
npm WARN audit fix string-width@3.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix cliui@5.0.0 node_modules/ganache-cli/node_modules/cliui
npm WARN audit fix cliui@5.0.0 is a bundled dependency of
npm WARN audit fix cliui@5.0.0 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix cliui@5.0.0 It cannot be fixed automatically.
npm WARN audit fix cliui@5.0.0 Check for updates to the ganache-cli package.
npm WARN audit fix yargs@13.2.4 node_modules/ganache-cli/node_modules/yargs
npm WARN audit fix yargs@13.2.4 is a bundled dependency of
npm WARN audit fix yargs@13.2.4 ganache-cli@6.12.2 at node_modules/ganache-cli
npm WARN audit fix yargs@13.2.4 It cannot be fixed automatically.
npm WARN audit fix yargs@13.2.4 Check for updates to the ganache-cli package.
npm WARN audit Updating ganache-cli to 6.5.0,which is a SemVer major change.

removed 56 packages, changed 29 packages, and audited 55 packages in 6s

# npm audit report

ansi-regex  >2.1.1 <5.0.1
Severity: moderate
 Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/ansi-regex
  strip-ansi  4.0.0 - 5.2.0
  Depends on vulnerable versions of ansi-regex
  node_modules/ganache-cli/node_modules/strip-ansi
    cliui  4.0.0 - 5.0.0
    Depends on vulnerable versions of strip-ansi
    node_modules/ganache-cli/node_modules/cliui
      yargs  8.0.0-candidate.0 - 15.0.0
      Depends on vulnerable versions of cliui
      Depends on vulnerable versions of os-locale
      Depends on vulnerable versions of yargs-parser
      node_modules/ganache-cli/node_modules/yargs
        ganache-cli  6.2.0-beta.0 - 6.5.0
        Depends on vulnerable versions of yargs
        node_modules/ganache-cli
    string-width  2.1.0 - 4.1.0
    Depends on vulnerable versions of strip-ansi
    node_modules/ganache-cli/node_modules/string-width

mem  <4.0.0
Severity: moderate
Denial of Service in mem - https://github.com/advisories/GHSA-4xcv-9jjx-gfj3
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/mem
  os-locale  2.0.0 - 3.0.0
  Depends on vulnerable versions of mem
  node_modules/ganache-cli/node_modules/os-locale
    yargs  8.0.0-candidate.0 - 15.0.0
    Depends on vulnerable versions of cliui
    Depends on vulnerable versions of os-locale
    Depends on vulnerable versions of yargs-parser
    node_modules/ganache-cli/node_modules/yargs
      ganache-cli  6.2.0-beta.0 - 6.5.0
      Depends on vulnerable versions of yargs
      node_modules/ganache-cli

y18n  <3.2.2
Severity: high
Prototype Pollution - https://github.com/advisories/GHSA-c4w7-xm78-47vh
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/y18n

yargs-parser  6.0.0 - 13.1.1
Severity: moderate
Prototype Pollution in yargs-parser - https://github.com/advisories/GHSA-p9pc-299p-vxgp
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/yargs-parser
  yargs  8.0.0-candidate.0 - 15.0.0
  Depends on vulnerable versions of cliui
  Depends on vulnerable versions of os-locale
  Depends on vulnerable versions of yargs-parser
  node_modules/ganache-cli/node_modules/yargs
    ganache-cli  6.2.0-beta.0 - 6.5.0
    Depends on vulnerable versions of yargs
    node_modules/ganache-cli

10 vulnerabilities (9 moderate, 1 high)

To address all issues, run:
  npm audit fix
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ npm audit fix
npm WARN audit fix ansi-regex@3.0.0 node_modules/ganache-cli/node_modules/ansi-regex
npm WARN audit fix ansi-regex@3.0.0 is a bundled dependency of
npm WARN audit fix ansi-regex@3.0.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix ansi-regex@3.0.0 It cannot be fixed automatically.
npm WARN audit fix ansi-regex@3.0.0 Check for updates to the ganache-cli package.
npm WARN audit fix y18n@3.2.1 node_modules/ganache-cli/node_modules/y18n
npm WARN audit fix y18n@3.2.1 is a bundled dependency of
npm WARN audit fix y18n@3.2.1 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix y18n@3.2.1 It cannot be fixed automatically.
npm WARN audit fix y18n@3.2.1 Check for updates to the ganache-cli package.
npm WARN audit fix yargs-parser@9.0.2 node_modules/ganache-cli/node_modules/yargs-parser
npm WARN audit fix yargs-parser@9.0.2 is a bundled dependency of
npm WARN audit fix yargs-parser@9.0.2 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix yargs-parser@9.0.2 It cannot be fixed automatically.
npm WARN audit fix yargs-parser@9.0.2 Check for updates to the ganache-cli package.
npm WARN audit fix mem@1.1.0 node_modules/ganache-cli/node_modules/mem
npm WARN audit fix mem@1.1.0 is a bundled dependency of
npm WARN audit fix mem@1.1.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix mem@1.1.0 It cannot be fixed automatically.
npm WARN audit fix mem@1.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix strip-ansi@4.0.0 node_modules/ganache-cli/node_modules/strip-ansi
npm WARN audit fix strip-ansi@4.0.0 is a bundled dependency of
npm WARN audit fix strip-ansi@4.0.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix strip-ansi@4.0.0 It cannot be fixed automatically.
npm WARN audit fix strip-ansi@4.0.0 Check for updates to the ganache-cli package.
npm WARN audit fix yargs@11.1.0 node_modules/ganache-cli/node_modules/yargs
npm WARN audit fix yargs@11.1.0 is a bundled dependency of
npm WARN audit fix yargs@11.1.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix yargs@11.1.0 It cannot be fixed automatically.
npm WARN audit fix yargs@11.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix os-locale@2.1.0 node_modules/ganache-cli/node_modules/os-locale
npm WARN audit fix os-locale@2.1.0 is a bundled dependency of
npm WARN audit fix os-locale@2.1.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix os-locale@2.1.0 It cannot be fixed automatically.
npm WARN audit fix os-locale@2.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix cliui@4.1.0 node_modules/ganache-cli/node_modules/cliui
npm WARN audit fix cliui@4.1.0 is a bundled dependency of
npm WARN audit fix cliui@4.1.0 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix cliui@4.1.0 It cannot be fixed automatically.
npm WARN audit fix cliui@4.1.0 Check for updates to the ganache-cli package.
npm WARN audit fix string-width@2.1.1 node_modules/ganache-cli/node_modules/string-width
npm WARN audit fix string-width@2.1.1 is a bundled dependency of
npm WARN audit fix string-width@2.1.1 ganache-cli@6.5.0 at node_modules/ganache-cli
npm WARN audit fix string-width@2.1.1 It cannot be fixed automatically.
npm WARN audit fix string-width@2.1.1 Check for updates to the ganache-cli package.

removed 9 packages, changed 29 packages, and audited 102 packages in 4s

2 packages are looking for funding
  run `npm fund` for details

# npm audit report

ansi-regex  >2.1.1 <5.0.1
Severity: moderate
 Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
fix available via `npm audit fix --force`
Will install ganache-cli@6.1.8, which is a breaking change
node_modules/ganache-cli/node_modules/ansi-regex
  strip-ansi  4.0.0 - 5.2.0
  Depends on vulnerable versions of ansi-regex
  node_modules/ganache-cli/node_modules/strip-ansi
    cliui  4.0.0 - 5.0.0
    Depends on vulnerable versions of strip-ansi
    Depends on vulnerable versions of wrap-ansi
    node_modules/ganache-cli/node_modules/cliui
      yargs  10.1.0 - 15.0.0
      Depends on vulnerable versions of cliui
      Depends on vulnerable versions of string-width
      node_modules/ganache-cli/node_modules/yargs
        ganache-cli  6.2.0-beta.0 - 6.12.2
        Depends on vulnerable versions of yargs
        node_modules/ganache-cli
    string-width  2.1.0 - 4.1.0
    Depends on vulnerable versions of strip-ansi
    node_modules/ganache-cli/node_modules/string-width
      wrap-ansi  3.0.0 - 6.1.0
      Depends on vulnerable versions of string-width
      Depends on vulnerable versions of strip-ansi
      node_modules/ganache-cli/node_modules/wrap-ansi

elliptic  <6.5.4
Severity: moderate
Use of a Broken or Risky Cryptographic Algorithm - https://github.com/advisories/GHSA-r9p9-mrjm-926w
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/elliptic

y18n  4.0.0
Severity: high
Prototype Pollution - https://github.com/advisories/GHSA-c4w7-xm78-47vh
fix available via `npm audit fix`
node_modules/ganache-cli/node_modules/y18n

9 vulnerabilities (8 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force
christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie networks modify ganache-ui host=http://172.29.64.1:7545
Brownie v1.17.0 - Python development framework for Ethereum

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/networks.py", line 68, in main
    fn(*args["<arguments>"])
  File "brownie/_cli/networks.py", line 151, in _modify
    raise ValueError(f"Network '{color('bright magenta')}{id_}{color}' does not exist")
ValueError: Network 'ganache-ui' does not exist

**Reply by cromewar** [reply]
Well I use ganache-ui just as an example, you should use the id you have configured, if you type `brownie networks list` you should see the networks there.

Also as you are using now the correct host name (WSL) you probably don´t need the `--network` modifier and just run it.

**Reply by ChristianG77** [reply]
Ok awesome we're making progress!  Got the ganache host sorted out, thanks for that!

Now last problem to sort out is I am getting an error saying "ModuleNotFoundError:  No module named 'home.christian'".

Seems I am not pointing my code to the right location because I do have "\\wsl$\Ubuntu-20.04\home\christian".

Is this where I need to change the hostname from "DESKTOP-PN8H8CK" to "wsl$"?

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ brownie run scripts/deploy_box.py
Brownie v1.17.0 - Python development framework for Ethereum

UpgradesProject is the active project.
Attached to local RPC client listening at '172.21.176.1:8545'...
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 53, in run
    module = _import_from_path(script)
  File "brownie/project/scripts.py", line 149, in _import_from_path
    _import_cache[import_str] = importlib.import_module(import_str)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
ModuleNotFoundError: No module named 'home.christian.'

**Reply by cromewar** [reply]
Is this where I need to change the hostname from "DESKTOP-PN8H8CK" to "wsl$"?
May be, I never tried this as I prefer to move my projects directly to wsl path, give it a try,

Also please delete your build folder and compile again, also make sure you are using the correct python version / environment

**Reply by ChristianG77** [reply]
Ok so if I understand correctly, it is to create and store the project folder structure in the WSL environment, so what I've done is sourced the code from here:  \\wsl$\Ubuntu-20.04\home\christian\.brownie\00_CODE\upgrades which then in the terminal takes inputs from christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$.

Using Python 3.8.10, I deleted the build, compiled again, ran scripts and received:  ModuleNotFoundError: No module named 'home.christian.'

I think the problem is in the bash (/home/christian) which is coming from \\wsl$\Ubuntu-20.04\... but DESKTOP-PN8H8CK might be pointing to Windows and not \\wsl$ because when I use WSL and source the code from Windows from here:  C:\Users\chris\Dropbox\00_CODE\upgrades which then in the terminal takes inputs from christian@DESKTOP-PN8H8CK:/mnt/c/Users/chris/Dropbox/00_CODE/upgrades$, it works both on Ganache and Rinkeby.

Is there an advantage to storing the code in the WSL environment or have I misunderstood something?

christian@DESKTOP-PN8H8CK:~/.brownie/00_CODE/upgrades$ ~
bash: /home/christian: Is a directory

**Reply by cromewar** [reply]
"Is there an advantage to storing the code in the WSL environment or have I misunderstood something?"

Well the main advantage is you wont struggle with conflicts on the hostname as you are doing now, Did you solve the problem? Any other thing I could help with?

**Reply by ChristianG77** [reply]
Thanks @cromewar .  So to make it work, would I need to change '@DESKTOP-PN8H8CK:' to '@wsl$'?

**Reply by cromewar** [reply]
Yes, give it a try, I simpler might be just move the project folders to WSL.

**Reply by ChristianG77** [reply]
I tried moving the project folders to WSL and loading VS Code using that path and received this error, 'ModuleNotFoundError: No module named 'home.christian.'

This is why I was wondering if the '@DESKTOP-PN8H8CK:' needed to change to '@wsl$'?

**Reply by cromewar** [reply]
That's estrange I've never faced a conflict with the hostnames, as you should select the interpreter on vscode which might be a virtualenv on your wsl path, please give a try changing the hostname, hope you can solve your problems.
